ocaml/tests/alerts/test_alert_certificate_check.ml

```diff
@@ -16,30 +16,33 @@ open Certificate_check
 
 let date_of = Xapi_stdext_date.Date.of_string
 
-let check_time = Xapi_stdext_date.Date.to_float (date_of "20200201T02:00:00Z")
+let check_time = date_of "20200201T02:00:00Z"
 
 let good_samples =
   [
     "20210202T02:00:00Z" (* +1 year*)
   ; "20200302T02:00:01Z" (* +30 days, +1 second *)
+  ; "20200302T02:00:00Z" (* +30 days *)
   ]
 
 let expiring_samples =
   [
-    ("20200302T02:00:00Z", Api_messages.host_server_certificate_expiring_30)
+    ("20200302T01:59:59Z", Api_messages.host_server_certificate_expiring_30)
   ; ("20200301T02:00:00Z", Api_messages.host_server_certificate_expiring_30)
-  ; ("20200215T02:00:00Z", Api_messages.host_server_certificate_expiring_14)
```

Comment:
It is because I see the last test case of `expiring_samples`:
`("20200201T02:00:00Z", Api_messages.host_server_certificate_expiring_07)`
as the check time is `20200201T02:00:00Z`, I think this means the "datetime" get from a certificate:
  let expiry =
    match cert with Host (_, exp) | Internal (_, exp) | CA (_, exp) -> exp
  in
is the last second when the certificate is valid, the certificate will become invalid from the next second of "expiry" (see the test case: `20200201T01:59:59Z` in `expired_samples`).

And I think the length of the "expiring 7 days" period should be exactly 7 days, while for case
`("20200208T02:00:00Z", Api_messages.host_server_certificate_expiring_07)`
the check_time(`20200201T02:00:00Z`) is still in "expiring 7 days" period after 7 days and 1 second, so I changed this test case to:
`("20200208T01:59:59Z", Api_messages.host_server_certificate_expiring_07)`
and added a case:
`("20200208T02:00:00Z", Api_messages.host_server_certificate_expiring_14)`

Reply:
It is because I see the last test case of `expiring_samples`:
`("20200201T02:00:00Z", Api_messages.host_server_certificate_expiring_07)`
as the check time is `20200201T02:00:00Z`, I think this means the "datetime" get from a certificate:
  let expiry =
    match cert with Host (_, exp) | Internal (_, exp) | CA (_, exp) -> exp
  in
is the last second when the certificate is valid, the certificate will become invalid from the next second of "expiry" (see the test case: `20200201T01:59:59Z` in `expired_samples`).

And I think the length of the "expiring 7 days" period should be exactly 7 days, while for case
`("20200208T02:00:00Z", Api_messages.host_server_certificate_expiring_07)`
the check_time(`20200201T02:00:00Z`) is still in "expiring 7 days" period after 7 days and 1 second, so I changed this test case to:
`("20200208T01:59:59Z", Api_messages.host_server_certificate_expiring_07)`
and added a case:
`("20200208T02:00:00Z", Api_messages.host_server_certificate_expiring_14)`